tests/unit/insertTestResult.unitTest.ts

```diff
@@ -1500,4 +1500,401 @@ describe("insertTestResult", () => {
                 });
         });
     });
+
+    // CVSB-7964: AC4
+    context("when inserting an PSV test result for LEC test code with mandatory fields: certificateNumber, expiryDate, modType, emissionStandard and fuelType populated for Pass tests", () => {
+        it("test record should get created and should not throw error", () => {
+            const testResult = testResultsPostMock[8];
+
+            MockTestResultsDAO = jest.fn().mockImplementation(() => {
+                return {
+                    createSingle: () => {
+                        return Promise.resolve(Array.of(testResultsPostMock[8]));
+                    },
+                    getTestNumber: () => {
+                        return Promise.resolve({ testNumber: "W01A00209", id: "W01", certLetter: "A", sequenceNumber: "002" });
+                    },
+                    getTestCodesAndClassificationFromTestTypes: () => {
+                        return Promise.resolve({
+                            linkedTestCode: "lcp",
+                            defaultTestCode: "lbp",
+                            testTypeClassification: "NON ANNUAL"
+                        });
+                    }
+                };
+            });
+
+            testResultsService = new TestResultsService(new MockTestResultsDAO());
+
+            return testResultsService.insertTestResult(testResult)
+                .then((insertedTestResult: any) => {
+                    expect(insertedTestResult[0].vehicleType).toEqual("psv");
+                    expect(insertedTestResult[0].testTypes[0].fuelType).toEqual("gas");
+                    expect(insertedTestResult[0].testTypes[0].emissionStandard).toEqual("0.08 g/kWh Euro 3 PM");
+                    expect(insertedTestResult[0].testTypes[0].certificateNumber).toEqual("8972gf");
+                    expect(insertedTestResult[0].testTypes[0].testExpiryDate).toEqual("2020-01-14");
+                    expect(insertedTestResult[0].testTypes[0].modType.code).toEqual("m");
+                    expect(insertedTestResult[0].testTypes[0].modType.description).toEqual("modification or change of engine");
+                });
+            });
+    });
+
+    context("when inserting a test result for LEC test code with unacceptable value for fuelType", () => {
+        it("should throw error", () => {
+            const testResult = testResultsPostMock[7];
```

Comment:
unnecessary line. can clone directly on next line, assuming "mutation of wrong object" comment above is correct.

Reply:
updated.